addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScannerUnitTest.java

```diff
@@ -0,0 +1,301 @@
+        // Given
+        HttpMessage msg = new HttpMessage();
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setRequestHeader("GET /some_action?q=" + jso + "&p=&m HTTP/1.1");
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInHeaderOfRequest() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setRequestHeader("GET / HTTP/1.1\r\n" + "X-Custom-Info: " + jso + "\r\n");
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInCookieOfRequest() throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setRequestHeader("GET / HTTP/1.1\r\n" + "Cookie: CRUNCHY=" + jso + "\r\n");
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInRawBodyOfPOSTRequest()
+            throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        byte[] jso = createJso();
+        msg.setRequestHeader(
+                "POST / HTTP/1.1\r\n"
+                        + "Content-Type: application/octet-stream\r\n"
+                        + "Content-Disposition: attachment; filename=\"jso.bin\"\r\n"
+                        + "Content-Length: "
+                        + jso.length
+                        + "\r\n");
+        msg.setRequestBody(jso);
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Ignore
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInBase64BodyOfRequest()
+            throws Exception {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        String jso = Base64.getEncoder().encodeToString(createJso());
+        msg.setRequestHeader(
+                "GET / HTTP/1.1\r\n"
+                        + "Content-Type: application/octet-stream\r\n"
+                        + "Content-Disposition: attachment; filename=\"jso.bin\"\r\n"
+                        + "Content-Length: "
+                        + jso.length()
+                        + "\r\n");
+        msg.setResponseBody(jso);
+
+        // When
+        rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+        // Then
+        assertThat(alertsRaised, hasSize(1));
+    }
+
+    @Ignore
+    @Test
+    public void shouldRaiseAlertGivenJsoMagicBytesAreDetectedInRequestParameterOfRequest()
```

Comment:
This seems to duplicate `shouldRaiseAlertGivenUriEncodedJsoMagicBytesAreDetectedInRequestParameterOfRequest`.